sources/megaproxylist.go

```diff
@@ -0,0 +1,127 @@
+package sources
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/nfx/slrp/pmux"
+	"github.com/rs/zerolog/log"
+)
+
+func init() {
+	Sources = append(Sources, Source{
+		ID:        65,
+		Homepage:  "https://www.megaproxylist.net",
+		Frequency: 24 * time.Hour,
+		Seed:      true,
+		Feed:      simpleGen(megaproxylist),
+	})
+}
+
+var megaproxylistUrl = fmt.Sprintf("https://www.megaproxylist.net/download/megaproxylist-csv-%s_SDACH.zip", time.Now().Format("20060102"))
+
+func unzipInMemory(body []byte) ([]byte, error) {
+	zipReader, err := zip.NewReader(bytes.NewReader(body), int64(len(body)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file as zip file: %w", err)
+	}
+
+	// Read all the files from zip archive
+	for _, zipFile := range zipReader.File {
+		if zipFile.Name != "megaproxylist.csv" {
+			continue
+		}
+
+		unzippedFileBytes, err := readZipFile(zipFile)
+		if err != nil {
+			return nil, fmt.Errorf("zip: can't read megaproxylist.csv")
+		}
+		return unzippedFileBytes, nil
+
+	}
+	return nil, fmt.Errorf("zip: can't find desired file")
+}
+
+func readZipFile(zf *zip.File) ([]byte, error) {
+	f, err := zf.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	return ioutil.ReadAll(f)
+}
+
+func getIpAddr(address string) (string, error) {
+	var addr string
+	if net.ParseIP(address) == nil {
```

Comment:
Does the test zip file have a localhost as addr, so that this function is unittested?